src/dotnet/commands/dotnet-add/dotnet-add-proj/Program.cs

```diff
@@ -104,9 +105,93 @@ private void AddProject(SlnFile slnFile, string projectPath)
                 };
 
                 slnFile.Projects.Add(slnProject);
+
+                EnsureBuildConfigurationsAreValidAndComplete(slnFile);
+
                 Reporter.Output.WriteLine(
                     string.Format(CommonLocalizableStrings.ProjectAddedToTheSolution, projectPath));
             }
         }
+
+        private void EnsureBuildConfigurationsAreValidAndComplete(SlnFile slnFile)
+        {
```

Comment:
this function is way too long, please split into smaller so it's clearer what it is doing